haystack/schema.py

```diff
@@ -729,23 +754,50 @@ def calculate_metrics(
             remarks: there might be a discrepancy between simulated reader metrics and an actual pipeline run with retriever top_k
-        :param doc_relevance_col: column in the underlying eval table that contains the relevance criteria for documents.
-            Values can be: 'gold_id_match', 'answer_match', 'gold_id_or_answer_match'.
-            Default value is 'gold_id_or_answer_match'.
         :param eval_mode: the input on which the node was evaluated on.
             Usually nodes get evaluated on the prediction provided by its predecessor nodes in the pipeline (value='integrated').
             However, as the quality of the node itself can heavily depend on the node's input and thus the predecessor's quality,
             you might want to simulate a perfect predecessor in order to get an independent upper bound of the quality of your node.
             For example when evaluating the reader use value='isolated' to simulate a perfect retriever in an ExtractiveQAPipeline.
             Values can be 'integrated', 'isolated'.
             Default value is 'integrated'.
+        :param document_scope: criterion for deciding whether documents are relevant or not.
+            You can select between:
+            - 'id': Document's id or custom id must match.
+                    Typical use case: Document Retrieval
+            - 'context': Document's content must match.
+                    Typical use case: Document-independent Passage Retrieval
+            - 'id_and_context': boolean operation `'id' AND 'context'`.
+                    Typical use case: Document-specific Passage Retrieval
+            - 'id_or_context': boolean operation `'id' OR 'context'`.
+                    Typical use case: Document Retrieval having sparse context labels
+            - 'answer': Document's content must include the answer. The selected `answer_scope` will be enforced.
+                    Typical use case: Question Answering
+            - 'id_or_answer' (default): boolean operation `'id' OR 'answer'`.
+                    This is intended to be a proper default value in order to support both main use cases:
+                    - Document Retrieval
+                    - Question Answering
+            Default value is 'id_or_answer'.
+        :param answer_scope: scope in which a matching answer is considered as correct.
+            You can select between:
+            - 'any' (default): any matching answer is considered as correct.
+                    For QA evalutions `document_scope` should be 'answer' or 'id_or_answer' (default).
+                    Select this for Document Retrieval and Passage Retrieval evaluations in order to use different `document_scope` values.
+            - 'context': answer is only considered as correct if its context matches as well.
+                    `document_scope` must be 'answer' or 'id_or_answer'.
+            - 'document': answer is only considered as correct if its document (id) matches as well.
```

Comment:
In `answer_scope` it's called `document` but in `document_scope` it's called `id`. However in both cases we are referring to something like `document_id`. Could we have a common name for both?

Reply:
Are you preferring `id` over `document_id` because of the flexibility that we allow with `custom_document_id_field`?

Reply:
Yes you're right: they basically mean the same thing: the document_id must match. document_id would be `custom_document_id` if `custom_document_id_field` is set and otherwise `Document.id`. 

However the consequence of setting document_scope to `id` is completely different than setting answer_scope to `document`: If they had the same name, people might expect it to be natural to set both to the same value (say to"document_id"). However that might be inconsistent as documents would only need to match the document_id whereas answers have to match the document_id AND the answer, ultimately grading the retriever better than it actual is. And that's what I tried to express by that. Unfortunately I wasn't very consistent on that end as "context" is named the same in both scopes and also has the same different consequences. :-(

So I tried to put that into the naming and just thought within the document_scope a "document_" prefix is superfluous as its rather obvious that the document's "id" is meant. For answer_scope it just felt natural to name the scope "document" within which the answer is only valid if it matches the document too.

Maybe I'm overthinking it a bit and `document_id` is the way to go. However then I wonder, how do we communicate that setting the answer_scope to the same value as document_scope is not supported. Do you think the ValueError we raise in that case is enough?

Reply:
I would prefer having the same name `document_id` in both scopes (just as with `context`). That this has different consequences in the different scopes is something we need to make clear in the documentation. The whole idea of context matching will need an own page in the documentation anyway in my opinion.